torchgeo/datasets/geo.py

```diff
@@ -511,11 +512,13 @@ def __init__(
                     )
                     raise AssertionError(msg)
 
-        if isinstance(res, float):
```

Comment:
@calebrob6 this was the bug causing the tutorial tests to fail.
```pycon
>>> isinstance(10, float)
False
>>> isinstance(10, int)
True
```
We need to check both int and float.